### ISSUE TITLE
Angular CLI v6: "Unknown option: '--environment'"

### ISSUE THREAD

**Issue by moloch--** [angular/angular-cli#10676]
### Versions
```
Angular CLI: 6.0.0
Node: 10.0.0
OS: darwin x64
Angular: 6.0.0
... animations, cli, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.6.0
@angular-devkit/build-angular     0.6.0
@angular-devkit/build-optimizer   0.6.0
@angular-devkit/core              0.6.0
@angular-devkit/schematics        0.6.0
@ngtools/webpack                  6.0.0
@schematics/angular               0.6.0
@schematics/update                0.6.0
rxjs                              6.1.0
typescript                        2.7.2
webpack                           4.7.0
```

### Repro steps

* Install Angular v6 / CLI v6
* `ng build --environment=foo`

### Observed behavior

The `build --environment` CLI option does not work in v6 (or was removed?), however the [documentation](https://github.com/angular/angular-cli/wiki/build) hasn't been updated to reflect the new way to build environments?

```
➜  web git:(ng-v6) ng build --environment=local
Unknown option: '--environment'
```

`ng build --help` Does not have any obvious replacement flags or information on how to specify environments during the build.

### Desired behavior
Any of:
 * Updated documentation
 * Backwards compatibility (and warnings for deprecated cli flags in prior versions) 
 * Fix `--environment`

### Mention any other details that might be useful (optional)

I know Angular CLI is a complicated project, but if you could push one update without breaking every existing project (and project build scripts/CI/CD) that rely on things like `--environment=` that'd be great, `ng update` doesn't seem to address things like this.

**Comment by moloch--** [comment]
It would still be nice to have an alias or warning about deprecated cli flag.s

**Comment by BBaysinger** [comment]
I'm getting this after upgrading to Angular 6, but I do not have `--environment` anywhere in my project. Thoughts?

**Comment by zijianhuang** [comment]
yes, the readme file updated 6 days ago is still using --env as example. please fix https://github.com/angular/angular-cli/wiki/build asap.

**Comment by logic01** [comment]
@BBaysinger, Do you have --env anywhere? That is an alias for --environment. There is a chance you have that somewhere in the scripts:{ } section of your package.json file.

**Comment by BBaysinger** [comment]
@logic01, I don’t have that anywhere either.

**Comment by Koesters** [comment]
ng build --aot --configuration=prod --output-hashing=all --sourcemaps=false --extract-css=true --named-chunks=false --build-optimizer=true

Configuration 'prod' could not be found in project 'sdsd'.
Error: Configuration 'prod' could not be found in project 'sdsd'.
    at Architect.getBuilderConfiguration (C:\sdsd\node_modules\@angular-devkit\architect\src\architect.js:106:23)
    at runSingleTarget (C:\sdsd\node_modules\@angular\cli\models\architect-command.js:138:89)
    at MergeMapSubscriber.rxjs_2.from.pipe.operators_1.concatMap.project (C:\sdsd\node_modules\@angular\cli\models\architect-command.js:143:127)
    at MergeMapSubscriber._tryNext (C:\sdsd\node_modules\rxjs\internal\operators\mergeMap.js:122:27)
    at MergeMapSubscriber._next (C:\sdsd\node_modules\rxjs\internal\operators\mergeMap.js:112:18)
    at MergeMapSubscriber.Subscriber.next (C:\sdsd\node_modules\rxjs\internal\Subscriber.js:103:18)
    at Observable._subscribe (C:\sdsd\node_modules\rxjs\internal\util\subscribeToArray.js:9:20)
    at Observable._trySubscribe (C:\sdsd\node_modules\rxjs\internal\Observable.js:177:25)
    at Observable.subscribe (C:\Users\Nils\src\deepship\node_modules\rxjs\internal\Observable.js:162:93)
    at MergeMapOperator.call (C:\sdsd\node_modules\rxjs\internal\operators\mergeMap.js:87:23)

**Comment by ranouf** [comment]
Same error than @Koesters 
Configuration 'prod' could not be found in project

**Comment by Koesters** [comment]
and sourcemaps became source-map 
But the whole command has not the same functionality.

There are also left over by the conversion script. In environment.ts it still refers to "_ng build --env=prod_" and '_.angular-cli.json_'

**Comment by codethief** [comment]
I arrived here because I, too, was greeted by `Unknown option: '--sourcemaps'` after updating to Angular 6.

How come none of these *breaking* changes are documented, neither in the [changelog](https://github.com/angular/angular-cli/releases) nor in the [wiki](https://github.com/angular/angular-cli/wiki/build)?!

**Comment by rtm** [comment]
Is there a complete, definitive, authoritative, up-to-date definition of the entire syntax and format of `angular.json`? I for one have not been able to lay my hands on it.

**Comment by ekowcharles** [comment]
@Koesters With CLI 6, these configuration exist as part of the `configuration` node like so

```json
"configurations": {
            "production": {
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "extractCss": true,
              "namedChunks": false,
              "aot": true,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true
            }
          }
```

In which case you need not pass them as arguments to the build task. Specify the `--configuration` flag should suffice.

**Comment by ekowcharles** [comment]
@rtm You will find the latest one [here](https://github.com/angular/angular-cli/wiki/angular-workspace).

**Comment by Reboog711** [comment]
I ended up here with a new install from scratch:

```
npm -g install @angular/cli
ng new my-app
cd my-app
ng serve -open
```

Console shows: Unknown option: '-p'.  

I am working in an empty directory and uninstalled @angular/cli before starting the new install.   

I'm also using Windows 10 and an elevated command prompt.

**Comment by clydin** [comment]
@Reboog711 `-open` and `--open` have different meanings.  `-open` is the equivalent of `-o -p -e -n`

**Comment by Reboog711** [comment]
@clydin  I had no idea; I was using a  single slash w/ the Angular CLI 5 and it worked as expected.  

The double slash does indeed address the errors.  Thanks!

**Comment by johnstsimoes** [comment]
This is disappointing. Folks, please consider backwards compatibility. Just renaming build parameters without at least deprecation warnings for a while is poor craftsmanship and raises concerns about NG being a first-tier choice.

**Comment by rtm** [comment]
@johnstsimoes Well, they didn't just randomly rename the build parameters. They changed the way configurations and environments are handled, making them more flexible and transparent. This necessitated the new `--configuration` flag; the chosen configuration then maps to instructions in `angular.json` relating to the envronment files. The deprecation notice warning that you want is in fact already there--not precisely a deprecation warning, but simply an error stating that there is no `--environment` option. Folks who run into this can easily enough Google "angular 6 environment" and figure things out quickly. If they bothered to open up the `angular.json` file that the upgrade tool created, they could also figure out it soon enough. If they read any Angular 6 migration guide, it would be covered there. Hence I find your comments and criticisms to be overly dramatic and exaggerated.

**Comment by unstephenk** [comment]
Switching to configuration setup in the angular.json using `ng serve -c=prod` causes the project to be extremely slow. Like its rebuilding every time I make a change. Is anyone else experiencing this sluggishness?

**Comment by rtm** [comment]
@unstephenk This is a new issue. Please open a new issue.

**Comment by KthProg** [comment]
Yeah guys! It wasn't randomly changed and it throws an error! Who cares if all your builds are suddenly breaking? To be fair, what other options are there? And it's a free tool so *shrug*

**Comment by ZQ-jhon** [comment]
good issue, thanks

**Comment by KthProg** [comment]
My comment was sarcastic FYI. I don't even remember what this issue was I work on mobile apps now lol.

**Comment by kkdg** [comment]
> It would still be nice to have an alias or warning about deprecated cli flag.s

I really don't like how `angular` treats its community
